pkg/reconciler/connection_pool_test.go

```diff
@@ -64,9 +65,9 @@ func mockValueMerger(old interface{}, new interface{}) interface{} {
 	return &merged
 }
 
-var serverConnectionPoolSetupTestCases = map[string]func(t *testing.T, ctx context.Context, opts ...ControlPlaneConnectionPoolOption) (control.Service, *ControlPlaneConnectionPool){
+var serverConnectionPoolSetupTestCases = map[string]func(t *testing.T, ctx context.Context, opts ...ControlPlaneConnectionPoolOption) (*network.ControlServer, *ControlPlaneConnectionPool){
```

Comment:
NIT: maybe later as part of a separate PR, having a var in a file between funcs just triggers me badly :smile: maybe move it to the top of the file?

Reply:
I want to burn these tests to the ground TBH :smile: I'm going to reorganize all these e2e tests in a separate package (maybe the `test` package introduced here https://github.com/knative-sandbox/control-protocol/pull/21). So we can have proper e2e tests without doing these bad tricks and a copy pasting good part of the initialization code